tempesta_fw/ss_skb.c

```diff
@@ -262,7 +305,7 @@ __new_pgfrag(struct sk_buff *skb, int size, int i, int shift, TfwStr *it)
 	 */
 	if (i == MAX_SKB_FRAGS) {
 		i = 0;
-		skb = it->skb;
+		skb = __next_skb_frag(skb, pskb);
```

Comment:
So instead of one write in ```__extend_pgfrags()``` and one read here you do 3 writes and 1 branch in ```__insert_skb_frag()``` and 1 read with a branch here. I don't think the change makes the code better.